src/main/java/com/lambdaworks/redis/protocol/ConnectionWatchdog.java

```diff
@@ -208,22 +206,23 @@ public synchronized void scheduleReconnect() {
             return;
         }
 
-        if ((channel == null || !channel.isActive()) && reconnectScheduleTimeout == null) {
+        if ((channel == null || !channel.isActive()) && !isWaitingReconnection) {
             attempts++;
 
             final int attempt = attempts;
             int timeout = (int) reconnectDelay.getTimeUnit().toMillis(reconnectDelay.createDelay(attempt));
```

Comment:
The synchronization with `isWaitingReconnection.compareAndSet(false, true)` could happen here, before obtaining the delay.